### PR TITLE
(Closes #2179) Fix LFRic eg14 and add compilation of tests and examples with nvfortran

### LINKED ISSUE #2179: LFRic eg14 won't build with nvfortran
Despite previous work on this (e.g. #1678), I can't currently build LFRic eg14 with nvfortran. (I can with gfortran, albeit ignoring the ACC directives.) The problem seems to be duplicate symbols at the link stage:

    nvlink error   : Multiple definition of '_fs_continuity_mod_21' in '../../../src/psyclone/tests/test_files/dynamo0p3/infrastructure/liblfric.a:fs_continuity_mod.o', first defined in 'main_psy.o'
    nvlink fatal   : merge_elf failed
    pgacclnk: child process exit status 2: /apps/packages/compilers/nvidia-hpcsdk/Linux_x86_64/23.5/compilers/bin/tools/nvdd

(The initial error said there were multiple definitions of `_argument_mod_21` so I deleted the pre-processed file `argument_mod.f90` before rebuilding and it gave me this error.) Perhaps the build system is building both the .f90 and .F90 files?

### PR DESCRIPTION
**Pull request by arporter**
Updates the (OpenACC) LFRic eg14 so that it works (builds and runs on GPU) with nvfortran.
I've also taken the opportunity to extend the work performed by the self-hosted runner.

### CODE REVIEW

**Comment by arporter** [comment]
Finally have the tests and examples working on the self-hosted runner. Ready for a first look from anyone!

**Comment by arporter** [comment]
Given that we had problems with the LFRic tutorials (#1833) it would be good if I extended this PR to also build the tutorials as part of the integration testing...

**Comment by arporter** [comment]
I've added building of the tutorials to the `compilation` workflow. This now means that the integration tests for this PR will fail until #'1833 is closed (#2172).

**Comment by arporter** [comment]
This is ready for review now but will require #2172 to be on before the updated integration tests will work. One for @TeranIvy, @rupertford or @hiker. Note that although I've made eg14 a bit more example, it still doesn't reproduce the tricky things that a real mini-app does (e.g. skeleton). I'll open a new issue to discuss how we're going to handle data movement.

**Comment by arporter** [comment]
#1833 is on master now so this is really ready for review :-)

File: examples/lfric/eg14/Makefile [hunk 1]
```diff
@@ -68,44 +81,50 @@ F90FLAGS += -I$(LFRIC_DIR)
...
+# invoke.
 %_psy.f90: %.x90
-	${PSYCLONE} -s ./acc_parallel_dm.py \
+	${PSYCLONE} -dm -s ./acc_parallel_dm.py  --profile invokes \
 	-opsy $*_psy.f90 -oalg $*_alg.f90 $<
 
 testkern_w0_kernel_0_mod.f90: main_psy.f90
 
-# TODO #1724 - compilation currently fails because module name in use
-# statement needs correcting following ACCRoutineTrans of Kernel.
```
[the remark this answers is not in the document]
Comment (arporter) [reply examples/lfric/eg14/Makefile:81]: Thanks for finding those. Hopefully all done now.

File: examples/lfric/eg14/acc_parallel_dm.py [hunk 2]
```diff
@@ -51,20 +52,33 @@ def trans(psy):
     transformed through the addition of a routine directive.
 
     '''
-    loop_trans = ACCLoopTrans()
-    parallel_trans = ACCParallelTrans()
+    const = LFRicConstants()
+
+    ctrans = Dynamo0p3ColourTrans()
+    # loop_trans = ACCLoopTrans()
+    # parallel_trans = ACCParallelTrans()
```
[the remark this answers is not in the document]
Comment (arporter) [reply examples/lfric/eg14/acc_parallel_dm.py:59]: All good points. I've tidied and re-named the script.

**Comment by arporter** [comment]
I've (hopefully) fixed the issue with the integration tests and have fired them off so will see how they did in the morning...

**Comment by sergisiso** [comment]
@arporter Note that in #2238 I also found the same module versioning issues and I opted to remove the version numbers altogether. Maybe its an easier solution to not having to change the workflow every time the modules on glados change

**Comment by arporter** [comment]
> Maybe its an easier solution to not having to change the workflow every time the modules on glados change

I'm just having this debate with @nmnobre :-) I think that the versions we use should be under our control and only changed as part of a PR. This means that the testing is then insulated from changes on glados (assuming that such changes don't remove modules entirely and without warning ;-) ).

**Comment by arporter** [comment]
I've discovered #2251 in the process of attempting to get the tutorials to build. I've therefore only enabled nvfortran compilation of the NEMO tutorials in this PR.

**Comment by arporter** [comment]
I finally have the compilation checks working. Ready for another look now.

**Review by sergisiso — APPROVED** [review]
@arporter Thanks for updating all the related documentation, all comments have been addressed. The new documentation builds fine (at least on my workstation, we will see on readthedocs...) and I can see that the compilation examples and tutorials now succeed on GHA. There aren't any remaining TODOs of this issue nor new formatting/linting warnings. This is ready to merge.

**Pull request merged**